cmd/goose/main.go

```diff
@@ -41,6 +41,10 @@ var (
 var version string
 
 func main() {
+	Main()
+}
+
+func Main() {
```

Comment:
This is typically not exported, and I'm not sure I want to allow a dependency on the CLI code. Note, there will be some changes to this coming very soon: https://github.com/pressly/goose/issues/663

I'm curious, what are you trying to do where the goose package isn't sufficient?

Reply:
hello there!

We want to embed go migrations in the goose binary.
Currently, this is done by implementing a custom CLI interface alongside the migrations, which is a bit annoying.

The laziest way to do it was to simply export the main function, so that is what I did. There are probably better ways to achieve this. The issue you linked looks promising :)

It would be really nice with a GooseCLI function or something that accepted configuration as input.

Reply:
Ah, okay I think I understand. You want to use Go migrations but also leverage the default CLI behaviour without re-implementing flags/args logic.